src/Microsoft.AspNet.OData.Shared/Extensions/ODataPathExtensions.cs

```diff
@@ -11,8 +11,16 @@
 
 namespace Microsoft.AspNet.OData.Extensions
 {
-    internal static class ODataPathExtensions
+    /// <summary>
+    /// Extensions method for <see cref="ODataPath"/>.
+    /// </summary>
+    public static class ODataPathExtensions
```

Comment:
Why is this changed from internal to public?

Reply:
For use by the customers using the nuget package

Reply:
Done in anticipation.
Also we are using them in E2E tests

Reply:
@KenitoInc E2E tests would hardly be a reason to make a method public. We use the friend library feature for that

Reply:
E2E tests mirror what the customer using the library would write the handler methods. That's why I still prefer keeping the methods public